src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -407,13 +247,13 @@ private static String createBuildLogFilePathFor(
                 coreCoords.version);
     }
 
-    private TestExecutionResult testPluginAgainst(
+    private void testPluginAgainst(
             MavenCoordinates coreCoordinates,
             UpdateSite.Plugin plugin,
             MavenRunner.Config mconfig,
             PomData pomData,
             PluginCompatTesterHooks pcth)
-            throws PluginSourcesUnavailableException, PomExecutionException, IOException {
+            throws PluginCompatibilityTesterException {
```

Comment:
Same pattern as before: `IOException`s are never allowed to propagate and must be immediately rethrown as `UncheckedIOException`. Any legitimate compatibility failure is thrown as a subclass of `PluginCompatibilityTesterException`.